### PR TITLE
feat: remove mark for deletion logic

### PR DESCRIPTION
**Pull request by ccalbaugh**
This PR removes the marking and subsequent deletion of nodes with no value during tree serialization.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>You reduced the total lines of code! Awesome! :+1:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 807506ba0c7805ae22a212f9e8ef796f56b79715
</p>

**Review by rma4ok — APPROVED** [review]
Looks good to me!

File: src/serialize.js [hunk 1]
```diff
@@ -24,10 +23,10 @@ export default (tree, { search } = {}) =>
         },
         x => {
           if (x.children) {
-            x.children = _.flow(
-              _.reject('markedForDeletion'),
-              _.reject(x => x.children && !x.children.length)
-            )(x.children)
+            x.children = _.reject(
+              x => x.children && !x.children.length,
+              x.children
```
Comment (daedalus28) [inline src/serialize.js:28]: While we're at it, I think we should just eliminate *all* node deletion by the client - leaving in the code that wipes empty groups still makes the returned tree not necessarily match up in index order

**Review by daedalus28 — CHANGES_REQUESTED** [review]

**Review by daedalus28 — CHANGES_REQUESTED** [review]
You also need to update the readme to remove mentions of this behavior. It's definitely in the "process algorithm" section and might be in other places

**Review by daedalus28 — APPROVED** [review]

**Pull request merged**